flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/delegation/Planner.java

```diff
@@ -65,7 +65,7 @@
 	 * @param operationPreConsumer pre consumer for specific operations
 	 * @return parsed queries as trees of relational {@link Operation}s
 	 */
-	List<Operation> parse(String statement, Consumer<Operation> operationPreConsumer);
+	Iterable<Operation> parse(String statement, Consumer<Operation> operationPreConsumer);
```

Comment:
I think once we have the `Iterable` return type, we don't need the Consumer anymore. The returned `Iterable` can be a lazy result (we should update the implementation of this method to adapt to return a lazy result).

Reply:
Sorry for the late reply, you're right, currently there's no consensus on this PR, I would keep it open until we really need this feature.